replikator-api.go

```diff
@@ -145,6 +138,22 @@ func startApiServer() {
 	}
 }
 
+func loggingMiddleware(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		log.Printf("[%s] %s %s", r.Method, r.RequestURI, r.RemoteAddr)
+
+		next.ServeHTTP(w, r)
+	})
+}
+func headerMiddleware(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
+		w.Header().Set("Access-Control-Allow-Origin", "*")
```

Comment:
For security it would be nice to set this with an envvar/config

Reply:
What would be the difference?

I'm not sure if there is a risk on setting the the CORS

Reply:
If you setup a website, somewhere, with a JS call to the replikator-api, send that URL to me, and I open it, it will do a request with my credentials. This is not secure. So I would not enable this by default.

Reply:
I've added a CORS secret with a min length of 20 chars. Only when this secret is set, and it's provided by the client the cors will be set.